test/e2e/ingress.go

```diff
@@ -224,16 +217,156 @@ func kubectlLogLBController(c *client.Client) {
+	// to ensure we test actual cluster functionality across upgrades.
+	rc.Spec = existingRc.Spec
+	rc.Name = "glbc"
+	rc.Namespace = cont.ns
+	rc.Labels = controllerLabels
+	rc.Spec.Selector = controllerLabels
+	rc.Spec.Template.Labels = controllerLabels
+	rc.Spec.Replicas = 1
+
+	// These command line params are only recognized by v0.51 and above.
+	testArgs := []string{
+		// Pass namespace uid so the controller will tag resources with it.
+		fmt.Sprintf("--cluster-uid=%v", cont.ns),
+		// Tell the controller to delete all resources as it quits.
+		fmt.Sprintf("--delete-all-on-quit=true"),
+		// Don't use the default Service from kube-system.
+		fmt.Sprintf("--default-backend-service=%v/%v", cont.svc.Namespace, cont.svc.Name),
+		// Only watch a single namespace, in case this test ends up running in parallel with others.
+		fmt.Sprintf("--watch-namespace=%v", cont.svc.Namespace),
+	}
+	for i, c := range rc.Spec.Template.Spec.Containers {
+		if c.Name == lbContainerName {
+			rc.Spec.Template.Spec.Containers[i].Args = append(c.Args, testArgs...)
+		}
+	}
+	cont.rc = rc
+	_, err = cont.c.ReplicationControllers(cont.ns).Create(cont.rc)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(waitForRCPodsRunning(cont.c, cont.ns, cont.rc.Name)).NotTo(HaveOccurred())
+}
+
+func (cont *ingressController) cleanup(del bool) error {
+	errMsg := ""
+	// Ordering is important here because we cannot delete resources that other
+	// resources hold references to.
+	fwList := []compute.ForwardingRule{}
+	gcloudUnmarshal("forwarding-rules", fmt.Sprintf("k8s-fw-.*--%v", cont.ns), &fwList)
+	if len(fwList) != 0 {
+		msg := ""
+		for _, f := range fwList {
+			msg += fmt.Sprintf("%v\n", f.Name)
+			if del {
+				Logf("Deleting forwarding-rule: %v", f.Name)
+				output, err := exec.Command("gcloud", "compute", "forwarding-rules", "delete",
+					f.Name, fmt.Sprintf("--project=%v", testContext.CloudConfig.ProjectID), "-q", "--global").CombinedOutput()
+				if err != nil {
+					Logf("Error deleting forwarding rules, output: %v\nerror:%v", string(output), err)
+				}
+			}
+		}
+		errMsg += fmt.Sprintf("\nFound forwarding rules:\n%v", msg)
+	}
+
+	tpList := []compute.TargetHttpProxy{}
+	gcloudUnmarshal("target-http-proxies", fmt.Sprintf("k8s-tp-.*--%v", cont.ns), &tpList)
+	if len(tpList) != 0 {
+		msg := ""
+		for _, t := range tpList {
+			msg += fmt.Sprintf("%v\n", t.Name)
+			if del {
+				gcloudDelete("target-http-proxies", t.Name)
+			}
+		}
+		errMsg += fmt.Sprintf("Found target proxies:\n%v", msg)
+	}
+
+	umList := []compute.UrlMap{}
+	gcloudUnmarshal("url-maps", fmt.Sprintf("k8s-um-.*--%v", cont.ns), &umList)
+	if len(umList) != 0 {
+		msg := ""
+		for _, u := range umList {
+			msg += fmt.Sprintf("%v\n", u.Name)
+			if del {
+				gcloudDelete("url-maps", u.Name)
+			}
+		}
+		errMsg += fmt.Sprintf("Found url maps:\n%v", msg)
+	}
+
+	beList := []compute.BackendService{}
+	gcloudUnmarshal("backend-services", fmt.Sprintf("k8s-be-[0-9]+--%v", cont.ns), &beList)
```

Comment:
Just curious why is the regex for backends and httphealthchecks `[0-9]+` while the other resources use `.*`? Also, can we depend on the namespace not being truncated or is there a check somewhere? Apologies if I missed it.